### PR TITLE
Fix channel tsv

### PR DESCRIPTION
**Pull request by kambysese**
@jasmainak added my pycharm housekeeping dir to ignore along with a few others I thought might come in handy later on. 
Also, some changes I thought would better align with BEP008 `*_channel.tsv` recommendations. LMKWYT.

### CODE REVIEW

File: mne_bids/bids_meg.py [hunk 1]
```diff
@@ -29,24 +29,29 @@ def _channel_tsv(raw, fname):
...
-
-    onlinefilter = '%0.2f-%0.2f' % (raw.info['highpass'], raw.info['lowpass'])
-    df = pd.DataFrame({'name': raw.info['ch_names'], 'type': ch_type,
-                       'description': description,
-                       'onlinefilter': onlinefilter,
-                       'samplingrate': '%f' % raw.info['sfreq'],
-                       'status': status
-                       })
+    low_cutoff, high_cutoff = (raw.info['highpass'], raw.info['lowpass'])
+    n = raw.info['nchan']
```
Comment (jasmainak) [inline mne_bids/bids_meg.py]: we normally don't use single letter variable names. Can you use something like `n_channels`?

File: mne_bids/bids_meg.py [hunk 2]
```diff
@@ -29,24 +29,29 @@ def _channel_tsv(raw, fname):
...
     map_chs = dict(grad='MEG', mag='MEG', stim='TRIG', eeg='EEG',
                    eog='EOG', misc='MISC')
-    map_desc = dict(grad='sensor gradiometer', mag='magnetometer',
-                    stim='analogue trigger',
-                    eeg='electro-encephalography channel',
-                    eog='electro-oculogram', misc='miscellaneous channel')
+    map_desc = dict(grad='Gradiometer', mag='Magnetometer',
+                    stim='Trigger',
+                    eeg='ElectroEncephaloGram',
+                    ecg='ElectroCardioGram',
```
Comment (jasmainak) [inline mne_bids/bids_meg.py]: I guess you would also have to update `map_chs`?

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,23 @@
...
+dist/
+*.egg*
 
+# Sphinx documentation
+docs/_build/
+doc/generated/
+doc/auto_examples/
+doc/auto_tutorials/
+doc/modules/generated/
+doc/sphinxext/cachedir
```
Comment (jasmainak) [inline .gitignore]: umm ... why did you add these to gitignore? These are for the docs

Comment (jasmainak) [inline .gitignore]: never mind this comment. It's good :)

File: mne_bids/bids_meg.py [hunk 4]
```diff
@@ -29,24 +29,29 @@ def _channel_tsv(raw, fname):
...
-                       'status': status
-                       })
+    low_cutoff, high_cutoff = (raw.info['highpass'], raw.info['lowpass'])
+    n = raw.info['nchan']
+    df = pd.DataFrame(columns=[['name', 'type', 'description',
+                                'sampling_frequency', 'low_cutoff',
+                                'high_cutoff', 'status']])
+    for col, data in zip(df.columns, [raw.info['ch_names'], ch_type,
+                                      description,
+                                      ['%.2f' % raw.info['sfreq']] * n,
```
Comment (jasmainak) [inline mne_bids/bids_meg.py]: if you use `pd.DataFrame`, you don't need this loop, do you?

**Comment by jasmainak** [comment]
Thanks @kambysese for carefully looking into the BEP008 specifications!

File: mne_bids/bids_meg.py (hunk 2 above)
Comment (kambysese) [inline mne_bids/bids_meg.py]: IDK according to BEP008 the recommended kw for `type` is `MEG` lets just go with that for now.

File: mne_bids/bids_meg.py (hunk 4 above)
Comment (kambysese) [inline mne_bids/bids_meg.py]: I was trying to pull off an ordered dict.

File: mne_bids/bids_meg.py [hunk 5]
```diff
@@ -29,24 +30,26 @@ def _channel_tsv(raw, fname):
...
-                       })
+    low_cutoff, high_cutoff = (raw.info['highpass'], raw.info['lowpass'])
+    n_channels = raw.info['nchan']
+    sfreq = raw.info['sfreq']
+    df = pd.DataFrame(OD([('name', raw.info['ch_names']),
+                          ('type', ch_type), ('discription', description),
+                          ('sampling_frequency', ['%.2f' % sfreq] * n_channels),
+                          ('low_cutoff', ['%.2f' % low_cutoff] * n_channels),
+                          ('high_cutoff', ['%.2f' % high_cutoff] * n_channels),
+                          ('status', status)]))
```
Comment (jasmainak) [inline mne_bids/bids_meg.py]: Sorry, last nitpick. Why not just do:

```py
df = df[['name', 'type', 'sampling_frequency', 'low_cutoff', 'high_cutoff', 'status']]
```

rather than use ordered dict?

File: mne_bids/bids_meg.py (hunk 2 above)
Comment (jasmainak) [inline mne_bids/bids_meg.py]: umm ... I'm not sure what you mean. This is what the draft reads:

> Recommended keywords for field type:
> MEG: MEG sensors 
> MEGREF: MEG reference sensors 
> EEG: ElectroEncephaloGram
> EOG: ElectroOculoGram (eyes)
> ECG: ElectroCardioGram (heart)
> EMG: ElectroMioGram (muscle) 
> TRIG: Triggers
> MISC: Miscellaneous

File: mne_bids/bids_meg.py (hunk 5 above)
Comment (kambysese) [inline mne_bids/bids_meg.py]: Well, that's why I'm getting myself into this mess 😄 ...to learn the ways. Will revert and fix.

File: mne_bids/bids_meg.py (hunk 2 above)
Comment (kambysese) [inline mne_bids/bids_meg.py]: I really wasn't sure if sensor types were to be declared in the `type` or `description` fields. Easy enough will do.

**Comment by kambysese** [comment]
@jasmainak are you okay with recent commit?

File: mne_bids/bids_meg.py (hunk 2 above)
Comment (jasmainak) [inline mne_bids/bids_meg.py]: okay looks like you misunderstood me and I misunderstood you :) I was saying that you have to update `map_chs` and add a key for ECG. I think you understood that I was asking you to change grad=MEG to grad=MAG !

**Comment by choldgraf** [comment]
Is this PR ready to go? I can merge if y'all are happy with it ;-)

**Comment by jasmainak** [comment]
Umm ... [this comment](https://github.com/mne-tools/mne-bids/pull/3/files#r127093580) hasn't yet been addressed yet. Yes, feel free to use your merge superpowers  :smiley:

**Comment by jasmainak** [comment]
Merged after cosmit https://github.com/mne-tools/mne-bids/commit/a5abf8c2630d3fa1413f540875fcd37c7e6874c7. @choldgraf I let you merge the next PR :)
